botocore/auth.py

```diff
@@ -154,7 +154,8 @@ def __init__(self, credentials, service_name, region_name):
         # We initialize these value here so the unit tests can have
         # valid values.  But these will get overriden in ``add_auth``
         # later for real requests.
-        self._region_name = region_name
+        self._signing_region = region_name
```

Comment:
Could we just keep these changes to the S3Sigv4 signer? I would prefer that this change did not affect all AWS services that use sigv4 signing.